### PR TITLE
Including users and groups when loading user_roles

### PR DESCRIPTION
**Pull request by danieljames-dj**
The user data and group data is used for post filtering, hence precomputing them for the roles list API.

### CODE REVIEW

File: app/controllers/api/v0/user_roles_controller.rb [hunk 1]
```diff
@@ -47,8 +50,8 @@ def index
     roles = UserRole.filter_roles(roles, current_user, params)
     roles = UserRole.sort_roles(roles, params[:sort])
 
-    # Limiting to first 100 elements of roles array to avoid serializing of large array.
-    render json: roles.first(100)
+    # Paginating the list by sending only first 100 elements unless mentioned in the API.
+    paginate json: roles
```
Comment (gregorbg) [inline app/controllers/api/v0/user_roles_controller.rb:54]: This change seems like it was included in your previous pagination PR.
Is there maybe a chance you didn't include the latest changes from `main` in your currentPR branch?

File: app/controllers/api/v0/user_roles_controller.rb [hunk 2]
```diff
@@ -19,6 +19,9 @@ class Api::V0::UserRolesController < Api::V0::ApiController
     group_id = params[:groupId]
     user_id = params[:userId]
 
+    # For post filtering, user & group is needed.
+    active_record = active_record.includes([:user, :group])
```
Comment (gregorbg) [inline app/controllers/api/v0/user_roles_controller.rb:23]: First, `includes` takes varargs, so `.includes(:user, :group)` will work.
Second, if you know that these two are going to be `includes`'d anyways no matter what, then you can just declare `active_record = UserRole.includes(...)` directly at the top of the method above.

**Review by gregorbg — CHANGES_REQUESTED** [review]

File: app/controllers/api/v0/user_roles_controller.rb (hunk 1 above)
Comment (danieljames-dj) [reply app/controllers/api/v0/user_roles_controller.rb]: Oops I don't know how this came, removed now.

File: app/controllers/api/v0/user_roles_controller.rb (hunk 2 above)
Comment (danieljames-dj) [reply app/controllers/api/v0/user_roles_controller.rb]: Oh ya, thanks for that. Implemented.

**Review by gregorbg — APPROVED** [review]

**Pull request merged**